src/main/java/com/connexta/store/poller/TransformStatusPoller.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.store.poller;
+
+import com.connexta.store.config.TransformConfiguration;
+import com.connexta.store.controllers.StoreController;
+import com.connexta.store.rest.models.AddMetadataRequest;
+import com.connexta.store.rest.models.MetadataInfo;
+import com.connexta.store.rest.models.QuarantineRequest;
+import com.connexta.transformation.rest.models.MetadataInformation;
+import com.connexta.transformation.rest.models.Status;
+import com.connexta.transformation.rest.models.TransformationPollResponse;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/** Class capable of polling the transformation poll endpoint. */
+@Slf4j
+public class TransformStatusPoller extends Thread {
+
+  private final WebClient storeWebClient;
+
+  private final WebClient transformWebClient;
+
+  private final BlockingQueue<TransformStatusTask> transformStatusQueue;
+
+  /** Creates a new TransformStatusPoller. */
+  public TransformStatusPoller(
+      BlockingQueue<TransformStatusTask> transformStatusQueue,
+      WebClient transformWebClient,
+      WebClient storeWebClient) {
+    this.transformStatusQueue = transformStatusQueue;
+    this.transformWebClient = transformWebClient;
+    this.storeWebClient = storeWebClient;
+  }
+
+  // TODO 11/20/2019 PeterHuffer: This class should be managed and be able to be properly destroyed
+  @SuppressWarnings("squid:S2189" /* Run forever by design */)
+  @Override
+  public void run() {
+    while (true) {
+      try {
+        doRun();
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+        return;
+      }
+    }
+  }
+
+  @VisibleForTesting
+  void doRun() throws InterruptedException {
+    TransformStatusTask transformStatusTask = transformStatusQueue.take();
+    boolean shouldRetry = handleTask(transformStatusTask);
+    if (shouldRetry) {
```

Comment:
:exclamation: It looks like there are cases where a transform request will never leave the queue. What happens if the endpoint's server has a problem and holds the request connection open forever? What happens if a job gets stuck in the IN_PROGRESS state and never advances?

Reply:
It would currently keep putting on the back of the queue, so it would cycle through, but some would get stuck in the queue. The quick solution right now would be to add a last tried timestamp and a failure attempt to the TransformStatusTask and only try a poll if 5 seconds have passed. And if we exceed X failures, just log it and remove the task completely.

Reply:
That would work. I think just the timestamp would be sufficient. No need to do both.

Reply:
I ended up adding back in the dynger polling library to use their retry and stop strategies.